plugins/modules/vmware_host_facts.py

```diff
@@ -282,6 +290,12 @@ def get_cluster_facts(self):
             cluster_facts.update(cluster=self.host.parent.name)
         return cluster_facts
 
+    def get_datacenter_facts(self):
+        datatacenter_facts = {'datacenter': None}
+        if self.host.parent and isinstance(self.host.parent.parent.parent, vim.Datacenter):
+            datatacenter_facts.update(datacenter=self.host.parent.parent.parent.name)
```

Comment:
If I understand your code correct, it would not always return the datacenter. It depends on `self.host.parent.parent.parent` beeing the datacenter... but what if the host (or cluster it's part of) is in a folder? I think this wouldn't work... although I'm not 100% sure there.

How about working all your way up until you reach `vim.Datacenter`? Maybe something like:

```
if self.host.parent:
    parent = self.host.parent
    while not isinstance(parent, vim.Datacenter):
        parent = parent.parent
    datatacenter_facts.update(datacenter=parent.name
```

I don't know if this will work, it's just a quick'n'dirty idea from me.

Reply:
I've thought about my suggestion and think it might run into a problem with stand-alone ESXi hosts. Maybe this would be better:

```
parent = self.host.parent
while parent:
    if isinstance(parent, vim.Datacenter):
        datatacenter_facts.update(datacenter=parent.name)
        break
    parent = parent.parent
```

@Nina2244 What do you think about this?

Reply:
I tested it and for our environment its the same. So I think it's better.